helix-term/src/ui/editor.rs

```diff
@@ -660,6 +660,17 @@ impl EditorView {
         let key_result = self.keymaps.get_mut(&mode).unwrap().get(event);
         self.autoinfo = key_result.sticky.map(|node| node.infobox());
 
+        // may cancel restore indent.
+        match &key_result.kind {
```

Comment:
Hmm I still don't like having this workaround inside key handling. Another option would be to check doc.changes() inside normal_mode, if it only contains a newline + indent

Reply:
Yeah, I agree that put to this location seems ugly...  I've checkout `doc.changes()`, but what if user open new line, then move cursor up/down, then press `esc` to goes into normal mode.  The doc doesn't change.

Reply:
I think that should be expected (and users generally shouldn't move around in insert mode).

Reply:
Done :-)  Check the dedent inside normal mode;

And for now, when user open a new line(whatever from `enter key` or from `open above/below` commands, normal mode will auto dedent(even user enter whitespaces after open a new line)

Personaly this behavior is better than I expected.  Can you check if this behavior is feasible?